go/clustermetadata/topo/locks.go

```diff
@@ -0,0 +1,373 @@
+// Copyright 2019 The Vitess Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Modifications Copyright 2025 Supabase, Inc.
+
+package topo
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"os"
+	"os/user"
+	"sync"
+	"time"
+
+	"github.com/spf13/pflag"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/multigres/multigres/go/common/mterrors"
+	"github.com/multigres/multigres/go/pb/mtrpc"
+	"github.com/multigres/multigres/go/servenv"
+	"github.com/multigres/multigres/go/tools/telemetry"
+)
+
+// This file contains utility methods and definitions to lock resources using topology server.
+
+// Lock describes a long-running lock on a database or a shard.
+// It needs to be public as we JSON-serialize it.
+type Lock struct {
+	// Action and the following fields are set at construction time.
+	Action   string
+	HostName string
+	UserName string
+	Time     string
+	Options  lockOptions
+
+	// Status is the current status of the Lock.
+	Status string
+}
+
+func init() {
+	servenv.OnParse(registerTopoLockFlags)
+}
+
+func registerTopoLockFlags(fs *pflag.FlagSet) {
+	fs.DurationVar(&RemoteOperationTimeout, "remote-operation-timeout", RemoteOperationTimeout, "time to wait for a remote operation")
```

Comment:
We have the opportunity to rename this if we want to. If we are importing this flag, we should start using it on all RPCs between multiorch and multipooler. That is how it is used in Vitess.

Reply:
I think this is a good idea. Should this be then more like a general timeout? It seems like this one shouldn't be in the topo package.